autoscale/src/main/java/com/sequenceiq/periscope/service/ScalingActivityService.java

```diff
@@ -40,6 +51,29 @@ public class ScalingActivityService {
     @Inject
     private RegionAwareCrnGenerator regionAwareCrnGenerator;
 
+    @Inject
+    private AutoscaleRestRequestThreadLocalService restRequestThreadLocalService;
+
+    @Inject
+    private ClusterService clusterService;
+
+    @Override
+    public AuthorizationResourceType getSupportedAuthorizationResourceType() {
+        return AuthorizationResourceType.DATAHUB;
+    }
+
+    @Override
+    @Retryable(value = NotFoundException.class, maxAttempts = 3, backoff = @Backoff(delay = 5000))
+    public String getResourceCrnByResourceName(String clusterName) {
+        return clusterService.findOneByStackNameAndTenant(clusterName, restRequestThreadLocalService.getCloudbreakTenant()).get().getStackCrn();
```

Comment:
What if the cluster hasn't been synced to periscope yet? The current implementation will result in an NPE, causing an authz failure. This logic should explicitly sync a cluster in case `ClusterService#findOneByStackNameAndTenant` returns an empty Optional, and throw a NotFoundException if CB isn't able to fetch the record. See `AutoScaleClusterCommonService#syncCBClusterByName`

Reply:
Done